tasks/install_shell.sh

```diff
@@ -434,13 +440,33 @@ do_perl() {
     unable_to_retrieve_package
   fi
 
-  # check for bad return status or empty output
-  if test $rc -ne 0 || test ! -s "$2"; then
-    capture_tmp_stderr "perl"
-    return 1
+  if test $rc -eq 0 && test -s "$2" ; then
+    return 0
   fi
 
-  return 0
+  capture_tmp_stderr "perl"
+  return 1
+}
+
+# do_perl_ff URL FILENAME
+do_perl_ff() {
+  info "Trying perl (File::Fetch)..."
+  run_cmd "perl -e 'use File::Fetch; use File::Copy; my \$ff = File::Fetch->new(uri => \$ARGV[0]); my \$outfile = \$ff->fetch() or die \$ff->server; copy(\$outfile, \$ARGV[1]) or die \"copy failed: \$!\"; unlink(\$outfile) or die \"delete failed: \$!\";' '$1' '$2' 2>>$tmp_stderr"
+  rc=$?
+
+  # check for 404
+  grep "HTTP response: 404" $tmp_stderr 2>&1 >/dev/null
+  if test $? -eq 0 ; then
```

Comment:
```suggestion
  if grep -q "HTTP response: 404" $tmp_stderr ; then
```

Reply:
There are so many things that could be improved in this script.
I generally kept the same format as the existing perl function to make it easy to review and approve.